src/components/FullDetailView/FullDetailPage.js

```diff
@@ -57,71 +67,118 @@ export class FullDetailPage extends Component {
+        this.setState({ changes: component.changes }, () => this.handleNewSpec(component))
+      } else this.handleNewSpec(component)
+    }
   }
 
   // Get the data for the current definition
   handleNewSpec(component) {
     const { token, uiInspectGetDefinition, uiInspectGetCuration, uiInspectGetHarvested } = this.props
-    if (component) {
-      uiInspectGetDefinition(token, component)
-      uiInspectGetCuration(token, component)
-      uiInspectGetHarvested(token, component)
-    }
+    if (!component) return false
+
+    uiInspectGetDefinition(token, component)
+    uiInspectGetCuration(token, component)
+    uiInspectGetHarvested(token, component)
+    this.previewDefinition(component)
   }
 
-  // Dispatch the action to save a contribution
+  /**
+   * Dispatch the action to save a contribution
+   * @param  {} description string that describes the contribution
+   */
   doContribute(description) {
-    const { token, component } = this.props
-    const patches = Contribution.buildContributeSpec({}, component)
+    const { token, component, curateAction } = this.props
+    const { changes } = this.state
+    const patches = Contribution.buildContributeSpec([], component, changes)
     const spec = { description: description, patches }
     curateAction(token, spec)
   }
 
+  // Action that calls the remote API that return a preview of the definition
+  previewDefinition(nextComponent) {
+    const { token, component, uiCurateGetDefinitionPreview } = this.props
+    const { changes } = this.state
+    if (!component && !nextComponent) return false
+
+    const previewComponent = nextComponent ? nextComponent : component
+    const patches = Contribution.buildPatch([], previewComponent, changes)
+    uiCurateGetDefinitionPreview(token, previewComponent, patches)
+  }
+
   // Shows the Modal to save a Contribution
   doPromptContribute() {
-    const { component } = this.props
-    if (!Contribution.hasChange(component)) return
+    const { changes } = this.state
+    if (isEmpty(changes)) return
     this.refs.contributeModal.open()
   }
 
-  handleSave(changes) {
-    const { onSave, component } = this.props
+  handleSave() {
+    const { onSave, component, uiCurateResetDefinitionPreview } = this.props
+    const { changes } = this.state
     const newComponent = { ...component, changes }
-    onSave && onSave(component, newComponent)
+    this.setState({ changes: {} }, () => {
+      uiCurateResetDefinitionPreview()
+      onSave && onSave(component, newComponent)
+    })
   }
 
   handleClose() {
     const { onClose } = this.props
-    onClose()
-    this.setState({ visible: false })
-  }
-
-  // Function called when a data has been changed
-  onChange(item, value) {
-    const { component } = this.props
     const { changes } = this.state
-    this.setState({ changes: Contribution.onChange(component, changes, item, value) }, () =>
-      console.log(this.state.changes)
+    if (isEmpty(changes)) return onClose()
```

Comment:
Break this out into a reuasable chunk of code (component or otherwise) then just call if from here.